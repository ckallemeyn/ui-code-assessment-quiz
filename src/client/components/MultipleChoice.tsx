import React, { useState, Fragment } from 'react'
import { FormField, Radio } from 'semantic-ui-react'

const MultipleChoice = ({ answers }: any) => {
  const [value, setValue] = useState("");
  const handleChange = (e: any, { value }: any): void => {
    e.preventDefault();
    return setValue(value);
  };


  return (
    <>
      {(answers.length > 0)
        ? answers.map((a: any) => (
            <Fragment key={a}>
              <FormField>
                <Radio
                  label={a}
                  name="radioGroup"
                  value={a}
                  checked={value === a}
                  onChange={handleChange}
                />
              </FormField>
            </Fragment>
          ))
        : null}
    </>
  );
}

export default MultipleChoice
