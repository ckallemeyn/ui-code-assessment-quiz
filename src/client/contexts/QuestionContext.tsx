import { createContext } from "react"

// maybe play around with creating an HOC for the entire app
const QuestionContext = createContext({});
QuestionContext.displayName = 'QuestionContext';

export default QuestionContext;
