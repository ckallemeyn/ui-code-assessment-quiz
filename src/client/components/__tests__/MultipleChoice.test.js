import React from 'react';
import { render } from '@testing-library/react';
import MultipleChoice from '../MultipleChoice';
import QuestionContext from '../../contexts/QuestionContext'

const renderWithContext = (value) => {

  return render(
    <QuestionContext.Provider value={value}>
      <MultipleChoice />
    </QuestionContext.Provider>
  )
}

let contextValue = {
  currentAnswer: '',
  handleChange: jest.fn(),
  state: {
    answers: ['4', '3', '2', '5']
  },
};


describe('Multiple Choice', () => {
  test('it should render', () => {
    const { container } = renderWithContext(contextValue);

    expect(container).toMatchSnapshot();
  });

  test('it should contain four different answer choices', () => {
    const { getAllByTestId, debug } = renderWithContext(contextValue);
    const radioNodeList = getAllByTestId('multiple-choice-answer')

    expect(radioNodeList.length).toBe(4);
    expect(radioNodeList[0].textContent).toBe('4');
    expect(radioNodeList[1].textContent).toBe('3');
    expect(radioNodeList[2].textContent).toBe('2');
    expect(radioNodeList[3].textContent).toBe('5');
  });

  test('it should not render when given an empty array of answers', () => {
    contextValue.state.answers = [];

    const { container } = renderWithContext(contextValue);

    expect(container).toBeEmptyDOMElement;
  });

});
