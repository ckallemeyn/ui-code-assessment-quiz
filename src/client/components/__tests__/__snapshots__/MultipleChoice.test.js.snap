// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple Choice it should render 1`] = `
<div>
  <div
    class="field"
    data-testid="multiple-choice-answer"
  >
    <div
      class="ui radio checkbox"
      data-testid="radio-btn"
    >
      <input
        class="hidden"
        name="radioGroup"
        readonly=""
        tabindex="0"
        type="radio"
        value="4"
      />
      <label>
        4
      </label>
    </div>
  </div>
  <div
    class="field"
    data-testid="multiple-choice-answer"
  >
    <div
      class="ui radio checkbox"
      data-testid="radio-btn"
    >
      <input
        class="hidden"
        name="radioGroup"
        readonly=""
        tabindex="0"
        type="radio"
        value="3"
      />
      <label>
        3
      </label>
    </div>
  </div>
  <div
    class="field"
    data-testid="multiple-choice-answer"
  >
    <div
      class="ui radio checkbox"
      data-testid="radio-btn"
    >
      <input
        class="hidden"
        name="radioGroup"
        readonly=""
        tabindex="0"
        type="radio"
        value="2"
      />
      <label>
        2
      </label>
    </div>
  </div>
  <div
    class="field"
    data-testid="multiple-choice-answer"
  >
    <div
      class="ui radio checkbox"
      data-testid="radio-btn"
    >
      <input
        class="hidden"
        name="radioGroup"
        readonly=""
        tabindex="0"
        type="radio"
        value="5"
      />
      <label>
        5
      </label>
    </div>
  </div>
</div>
`;
